src/core/configuration.ts

```diff
@@ -10,6 +10,7 @@ interface ExpectedChild {
 
 class Configuration {
 
+    public changedOptions: any;
```

Comment:
I think it's not the best place for the `changedOptions`. (See comments below)

What is more.. how about renaming it to smth like `innerChanges` or `internalChanges` or `mutedChanges`